### DISCUSSION TITLE
C++17 Automotive Coding Standards

### DISCUSSION THREAD

**Discussion by gpalmer-latai** [eclipse-iceoryx/iceoryx#2075] [General]
Hi!

With the proposed upgrade to C++17, I was curious what peoples thoughts/stories are about using Iceoryx in Codebases that need to adhere to some automotive C++ standards.

Currently, the only officially published standards are AUTOSAR C++14 and MISRA 2008. MISRA will apparently come out with some C++17 standards any day now (for the past couple of years), but there is still no official word when. I wonder what other folk's approaches are for asserting that their code lives up to some standard.

**Comment by elBoberido** [comment]
I'm not quite sure but I think many tools already support the latest draft for MISRA C++17 rules. I guess that's what most of them will use.

@mossmaurice AFAIK you had some contact with Peter Sommerlad. Is there something you can share?

**Reply by gpalmer-latai** [reply]
Is the latest draft available somewhere? 

Do you know any/which tools supposedly are based on it?

**Reply by elBoberido** [reply]
I don't know if the draft is already available but it seems [Helix QAC](https://www.perforce.com/products/helix-qac/whats-new-helix-qac) already integrated the rules with the latest release and Axivion [will](https://www.axivion.com/en/products/coding-guidelines/misra-check/) [soon](https://www.qt.io/product/quality-assurance/axivion) have them too.

@sbellon do you have an ETA for MISRA C++17 coding rules in Axivion?

**Reply by sbellon** [reply]
Axivion Suite has incorporated rules from the upcoming MISRA-C++202x standard already starting with release 7.5.0 one year ago. Over the time we have added more rules and with the upcoming release 7.7.0 we will have ~2/3 rule coverage and will continue to expand until we have full coverage.

**Reply by elBoberido** [reply]
@sbellon thanks for the quick response

**Reply by mossmaurice** [reply]
@gpalmer-latai 

> Currently, the only officially published standards are AUTOSAR C++14 and MISRA 2008. MISRA will apparently come out with some C++17 standards any day now (for the past couple of years), but there is still no official word when.

I was part of the official reviewer group for MISRA-C++:202x which took place in Q3/Q4 2022. To me, it was already in a solid state, considering and referring to C++17 and C++20 features on several occasions and hence much better suited for projects like iceoryx using modern C++ than the previous rulesets or the AUTOSAR one :heart: 

> Is the latest draft available somewhere?

Similarly to the previous MISRA rulesets, I suppose this is not freely available due to licensing topics. I think you need to contact the [MISRA org](https://misra.org.uk/), but INAL.

@elBoberido 
 
> AFAIK you had some contact with Peter Sommerlad. Is there something you can share?

No, I did not (yet). My latest status is the following from the [article he wrote on heise.de](https://www.heise.de/hintergrund/MISRA-C-bietet-Richlinien-und-Konformitaet-auch-fuer-neuen-Sprachstandard-C-20-4980750.html?seite=all) end of 2020. Let me just ping him here :-)

@PeterSommerlad Thanks for your great work on the MISRA-C++:202x ruleset! Do you have an ETA for its release or could provide a contact to ask (possibly via a private message)?

**Reply by gpalmer-latai** [reply]
Thanks for the feedback! This is all great information.

**Reply by sbellon** [reply]
Just to follow up: Axivion Suite 7.7 with support for the initial 2/3 of the MISRA-C++:2023 rules is released now, full coverage to be expected in release 7.8.

**Comment by PeterSommerlad** [comment]
Hi all,

wrt to MISRA-C++:2023 (addressing C++17) the release is really around
the corner. You can monitor their shop https://misra.org.uk/shop/ when
the document becomes available. It really should happen within the next
two weeks afaik. While behind a paywall, the price is reasonable enough
to get a PDF (GBP15) and even the hard-copies are sold by
print-on-demand for more or less cost-coverage (MISRA-C is GBP45).

Regards
Peter.

Simon Hoinkis wrote on 07.11.23 11:07:
> @gpalmer-latai <https://github.com/gpalmer-latai>
> 
>     Currently, the only officially published standards are AUTOSAR C++14
>     and MISRA 2008. MISRA will apparently come out with some C++17
>     standards any day now (for the past couple of years), but there is
>     still no official word when.
> 
> I was part of the official reviewer group for MISRA-C++:202x which took
> place in Q3/Q4 2022. To me, it was already in a solid state, considering
> and referring to C++17 and C++20 features on several occasions and hence
> much better suited for projects like iceoryx using modern C++ than the
> previous rulesets or the AUTOSAR one ❤️
> 
>     Is the latest draft available somewhere?
> 
> Similarly to the previous MISRA rulesets, I suppose this is not freely
> available due to licensing topics. I think you need to contact the MISRA
> org <https://misra.org.uk/>, but INAL.
> 
> @elBoberido <https://github.com/elBoberido>
> 
>     AFAIK you had some contact with Peter Sommerlad. Is there something
>     you can share?
> 
> No, I did not (yet). My latest status is the following from the article
> he wrote on heise.de
> <https://www.heise.de/hintergrund/MISRA-C-bietet-Richlinien-und-Konformitaet-auch-fuer-neuen-Sprachstandard-C-20-4980750.html?seite=all>
> end of 2020. Let me just ping him here :-)
> 
> @PeterSommerlad <https://github.com/PeterSommerlad> Thanks for your
> great work on the MISRA-C++:202x ruleset! Do you have an ETA for its
> release or could provide a contact to ask (possibly via a private message)?
> 
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse-iceoryx/iceoryx/discussions/2075#discussioncomment-7497134>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANFLFKGOWQOTAEE6I5MSFDYDIB7HAVCNFSM6AAAAAA63LSOJOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIOJXGEZTI>.
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> 


-- 
Peter Sommerlad

Better Software: Consulting, Training, Reviews
Modern, Safe & Agile C++

***@***.***
+41 79 432 23 32

**Reply by elBoberido** [reply]
@PeterSommerlad thanks very much
